packages/pool/contracts/StakeUtils.sol

```diff
@@ -81,43 +77,6 @@ contract StakeUtils is TransferUtils, IStakeUtils {
             userStakedNow >= amount,
             ERROR_VALUE
             );
-
-        // Revoke the reward of the current epoch if applicable
-        uint256 currentEpoch = block.timestamp / EPOCH_LENGTH;
-        if (!user.epochIndexToRewardRevocationStatus[currentEpoch])
-        {
-            Reward storage currentReward = epochIndexToReward[currentEpoch];
-            if (currentReward.amount != 0)
-            {
-                uint256 tokensToRevoke = currentReward.amount
-                     * getValueAt(user.shares, currentReward.atBlock)
-                     / getValueAt(totalShares, currentReward.atBlock);
```

Comment:
`totalShares` is expected to grow quite large, so calling `getValueAt()` on it is something we really want to avoid